### ISSUE TITLE
Included build artifacts substituted into buildscript classpath should be constructed in parallel

### ISSUE THREAD

**Issue by bigdaz** [gradle/gradle#2518]
@bigdaz commented on [Mon May 08 2017](https://github.com/gradle/composite-builds/issues/105)

Don't use build-on-demand for plugin resolution, instead introduce a separate task execution phase for building the plugin classpath.
